slynk/backend/allegro.lisp

```diff
@@ -1083,3 +1083,26 @@ to do this, this factors in the length of the inserted header itself."
 
 (defimplementation wrapped-p (spec indicator)
   (getf (excl:fwrap-order (process-fspec-for-allegro spec)) indicator))
+
+
+(defimplementation wait-for-input (streams &optional timeout)
+  (mp:wait-for-input-available streams
+                               :timeout (unless (eq t timeout)
+                                          timeout)))
+
+
+(defvar *fd-handler* nil)
```

Comment:
I haven't tried this out yet, but I'd say this simple boolean flag should be replaced by a per-socket list of handlers, so that SLY can add and remove multiple fd handlers to multiple sockets.  A hash table `*fd-handlers*` should suffice.